climada/util/dwd_icon_loader.py

```diff
@@ -0,0 +1,258 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU Lesser General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
```

Comment:
Just as a general thing; should we create an entirely new module for the dwd stuff? 

Or should we make a module that accumulates ALL kinds of file download functions? (e.g., i might have similar url-creation functions for OSM data queries, etc etc., so I might eventually put them in a downloader module, too.

Reply:
yeah, i don't know if utils is the place to put it, if it's only used in the StormEurope module
i quite like the idea of a module for download functions
though deciding this shouldn't stop us from merging the PR

Reply:
though this might be a Big Conversation since it's different to `tc_tracks`' way of handling data

Reply:
I hope this functionality can be used outside of StormEurope, e.g. for Heat or Rainfall, that is why i put it in util. I created a new module, because there is very specific functions for this particular provider as the files/filenames change each day (would be similar with OSM probably). For the download of a few static files, I would not recommend building a specific module. Is it ok to merge this PR like this?

Reply:
I don't have a strong opinion on this. Slight preference for a general download / reader module. Naming a very explicit module (such as the `dwd_icon_loader`) makes it of course easier to infer the specific functions that can be found inside as opposed to having to look through a `data_download` module and see if your case is covered. On the other hand, having lots of modules (`loader_x`, `loader_y`, `download_file_z`) etc. make the util folder quite crowded..

Reply:
ahhh, that makes sense. i'm fine with merging as it is.

maybe extend the comment at the start of the file saying it's currently used for the DWD wind forecast but will easily adapt to work with other variables.

Reply:
Comment is extended